### PR TITLE
Variable aggregation

### PR DESCRIPTION
**Pull request by filiphr**
Add support for BPMN Multi Instance and CMMN Repetition variable aggregation

### CODE REVIEW

File: modules/flowable-cmmn-converter/src/main/java/org/flowable/cmmn/converter/CmmnXmlConverter.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import java.io.InputStreamReader;
 import java.io.OutputStreamWriter;
 import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-converter/src/main/java/org/flowable/cmmn/converter/CmmnXmlConverter.java:22]: Import is not used.

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/TaskHelper.java [hunk 2]
```diff
@@ -18,7 +18,10 @@
 import java.util.Map;
 
 import org.apache.commons.lang3.StringUtils;
+import org.flowable.bpmn.model.FlowElement;
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/TaskHelper.java:21]: Import not used

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/TaskHelper.java [hunk 3]
```diff
@@ -18,7 +18,10 @@
...
+import org.flowable.bpmn.model.UserTask;
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/TaskHelper.java:22]: Import not used.

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/TaskHelper.java [hunk 4]
```diff
@@ -18,7 +18,10 @@
...
 import org.flowable.common.engine.api.FlowableException;
+import org.flowable.common.engine.api.delegate.Expression;
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/TaskHelper.java:24]: Import not used

**Comment by filiphr** [comment]
Good catch @dbmalkovsky. Thanks for the info

File: modules/flowable-cmmn-api/src/main/java/org/flowable/cmmn/api/delegate/PlanItemVariableAggregator.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+     * @param context the aggregation context
+     * @return the value for the aggregated variable
+     */
+    Object aggregateSingleVariable(DelegatePlanItemInstance planItemInstance, PlanItemVariableAggregatorContext context);
+
+    /**
+     * Aggregated the provide variable instances into one variable value.
+     *
+     * This is called when all repeatable plan item instances are complete, or an overview for repeatable plan item insatnces is needed.
+     *
```
Comment (im-pratham) [inline modules/flowable-cmmn-api/src/main/java/org/flowable/cmmn/api/delegate/PlanItemVariableAggregator.java:41]: Typo in instances.

Comment (filiphr) [reply modules/flowable-cmmn-api/src/main/java/org/flowable/cmmn/api/delegate/PlanItemVariableAggregator.java:41]: Good catch @im-pratham.  I've fixed this in https://github.com/flowable/flowable-engine/commit/3fb4edcf85ba7f16b6a7c1345515fe45c802628d

**Pull request merged**